crates/ball_filter/src/hypothesis.rs

```diff
@@ -17,104 +20,132 @@ use types::{
 pub mod moving;
 pub mod resting;
 
+#[derive(Clone, Debug, Serialize, Deserialize, PathSerialize, PathDeserialize, PathIntrospect)]
+pub enum BallMode {
+    Resting(MultivariateNormalDistribution<2>),
+    Moving(MultivariateNormalDistribution<4>),
+}
+
 #[derive(Clone, Debug, Serialize, Deserialize, PathSerialize, PathDeserialize, PathIntrospect)]
 pub struct BallHypothesis {
-    pub moving: MultivariateNormalDistribution<4>,
-    pub resting: MultivariateNormalDistribution<2>,
+    pub mode: BallMode,
     pub last_seen: SystemTime,
     pub validity: f32,
 }
 
 impl BallHypothesis {
-    pub fn new(
-        moving_hypothesis: MultivariateNormalDistribution<4>,
-        resting_hypothesis: MultivariateNormalDistribution<2>,
-        last_seen: SystemTime,
-    ) -> Self {
+    pub fn new(hypothesis: MultivariateNormalDistribution<4>, last_seen: SystemTime) -> Self {
         Self {
-            moving: moving_hypothesis,
-            resting: resting_hypothesis,
+            mode: BallMode::Moving(hypothesis),
```

Comment:
because only the state transition moving to resting is implemented using the kalman filter (on purpose, since the transition ready to moving is not continuous in the real world)